store/mockstore/mocktikv/mvcc_leveldb.go

```diff
@@ -931,6 +931,46 @@ func (mvcc *MVCCLevelDB) Cleanup(key []byte, startTS uint64) error {
 	return mvcc.db.Write(batch, nil)
 }
 
+// TxnHeartBeat implements the MVCCStore interface.
+func (mvcc *MVCCLevelDB) TxnHeartBeat(key []byte, startTS uint64, adviseTTL uint64) (uint64, error) {
+	startKey := mvccEncode(key, lockVer)
+	iter := newIterator(mvcc.db, &util.Range{
+		Start: startKey,
+	})
+	defer iter.Release()
+
+	if iter.Valid() {
+		dec := lockDecoder{
+			expectKey: key,
+		}
+		ok, err := dec.Decode(iter)
+		if err != nil {
+			return 0, errors.Trace(err)
+		}
+		if ok && dec.lock.startTS == startTS {
+			lock := dec.lock
+			batch := &leveldb.Batch{}
+			// Increase the ttl of this transaction.
+			if adviseTTL < lock.ttl {
+				lock.ttl = lock.ttl/2 + adviseTTL/2
```

Comment:
It can decrease?😲

Reply:
It doesn't matter... `adviseTTL` is just advice from TiDB.
TiKV could decide how to handle the TTL itself.

The TTL of a large transaction is pre-calculated (estimated), and maybe the value is too large,
or, TTL is set to math.MaxUint64 by mistake ... decrease it doesn't affect the correctness of 2PC.

Reply:
@youjiali1995

Reply:
Why not simply `lock.ttl = adviseTTL`?

Reply:
I think it would be simpler if we keep the ttl increasing monotomically.
The old `TxnHeartBeat` maybe delay to send, it may cause the transaction to rollback.

Reply:
If we can update the TTL, then we can use a max initial TTL in case it is too large.